### ISSUE TITLE
Element doesn't work well with native Wayland

### ISSUE THREAD

**Issue by jplatte** [vector-im/element-web#18943]
Changelog: https://www.electronjs.org/docs/breaking-changes#planned-breaking-api-changes-140

I've just tried running Element in native Wayland mode (`--enable-features=UseOzonePlatform --ozone-platform=wayland`) and it seems that currently breaks Element entirely in smoe circumstances (had the window be fully transparent at startup once, and in general it goes all white when trying to join a Jitsi conference). All my apps using Electron 14 as well as Chromium seem to run fine with these flags so I hope once Element upgrades it will work too :)

Being able to run Element natively under Wayland would have many advantages, most notably proper HiDPI support in wayland-based desktop environments.

**Comment by wantija** [comment]
Updating to Electron 14 will allow (hopefully) the desktop app to use KDE's file picker when uploading files with `GTK_USE_PORTAL=1` exported, hope this happens soon!

**Comment by dbkr** [comment]
Electron 14 pulls in chrome 93 which causes problems with jitsi. We'll need to make sure these are fixed before updating.

**Comment by jplatte** [comment]
Where can I read more about these issues with Jitsi? Could an upgrade to Electron v15 be done directly, which is based on Chromium 94?

**Comment by novocaine** [comment]
I *think* there is a known workaround noted in https://github.com/vector-im/element-web/issues/18943#issuecomment-938748142 so flagging as S-minor (although I guess for those who don't know the non-obvious workaround, the issue is critical)

**Comment by novocaine** [comment]
@jplatte - how confident are you that upgrading to Electron 14 will actually solve the Wayland issues? Are you just basing that off other Electron 14 apps seeming okay? I couldn't quickly find anything specific about Electron 14 that suggests its Wayland support is improved - I worry Element just has some specific issues that need be worked through that aren't linked to Electron version.. 

If you're game for testing it on your system, you could try [flipping this line to 14 locally](https://github.com/vector-im/element-desktop/blob/develop/package.json#L60) and taking the build for a spin..

**Comment by jplatte** [comment]
> Are you just basing that off other Electron 14 apps seeming okay?

Yes

> I couldn't quickly find anything specific about Electron 14 that suggests its Wayland support is improved

Have you checked the corresponding Chromium changelogs? 

> If you're game for testing it on your system, you could try [flipping this line to 14 locally](https://github.com/vector-im/element-desktop/blob/develop/package.json#L60) and taking the build for a spin..

Sure, will try tomorrow.

**Comment by novocaine** [comment]
> > Are you just basing that off other Electron 14 apps seeming okay?
> 
> Yes
> 
> > I couldn't quickly find anything specific about Electron 14 that suggests its Wayland support is improved
> 
> Have you checked the corresponding Chromium changelogs?

I have not! Can you link to what you've noticed?

> 
> > If you're game for testing it on your system, you could try [flipping this line to 14 locally](https://github.com/vector-im/element-desktop/blob/develop/package.json#L60) and taking the build for a spin..
> 
> Sure, will try tomorrow.

Awesome :)

**Comment by damnms** [comment]
the problem i reported is not linked to wayland(!), this problem occurs on _all_ fedora 35 versions, no matter which x window system is used.
this problem also affected ms teams, signal, etc. - but they already provided updates.
teams released an update that fixed it correctly, the signal version for fedora just added the mentioned start parameter.
i only posted here because i got "redirected" from my original bug report ;)

**Comment by jplatte** [comment]
So I changed that line, built and ran with `./node_modules/.bin/electron . --enable-features=UseOzonePlatform --ozone-platform=wayland`. It didn't even start (it did without the flags), instead raising the following errors:

```
[16760:1116/113754.825514:ERROR:cursor_loader.cc(114)] Failed to load a platform cursor of type kNull
[16795:0100/000000.864680:ERROR:gpu_init.cc(457)] Passthrough is not supported, GL is egl, ANGLE is
/home/jplatte/code/element-desktop/node_modules/electron/dist/electron exited with signal SIGSEGV
[16795:0100/000000.647927:ERROR:sandbox_linux.cc(376)] InitializeSandbox() called with multiple threads in process gpu-process.
```

**Comment by jplatte** [comment]
Oh, actually `yarn add electron` from the readme seems to have overwritten the electron version to 16.0.0 ...

**Comment by jplatte** [comment]
Starting from git w/o a config directory, or only at the old `Riot` path results in

> Your Element is misconfigured
>
> Invalid configuration: no default server specified.

Not going to open a separate issue because the issue template looks super complicated ._.

**Comment by jplatte** [comment]
Alright so I managed to run it with native wayland & electron 14. The close confirmation prompt bug remains and the mouse cursor is tiny when on the window (same in regular Chromium, there's a reason using this still requires cmdline flags), but otherwise it seems usable. Notably, Jitsi works¹.

¹ I added a widget w/ the integration manager; the conference name was literally `$roomName` but that seems like an unrelated bug

**Comment by novocaine** [comment]
> Alright so I managed to run it with native wayland & electron 14. The close confirmation prompt bug remains and the mouse cursor is tiny when on the window (same in regular Chromium, there's a reason using this still requires cmdline flags), but otherwise it seems usable. Notably, Jitsi works¹.
> 
> ¹ I added a widget w/ the integration manager; the conference name was literally `$roomName` but that seems like an unrelated bug

Thanks so much for getting this working and testing. Sounds promising, we will see what testing produces on other platforms.

**Comment by novocaine** [comment]
> the problem i reported is not linked to wayland(!), this problem occurs on _all_ fedora 35 versions, no matter which x window system is used. this problem also affected ms teams, signal, etc. - but they already provided updates. teams released an update that fixed it correctly, the signal version for fedora just added the mentioned start parameter. i only posted here because i got "redirected" from my original bug report ;)

Okay - https://github.com/vector-im/element-web/issues/19743#issuecomment-968078069 suggests that @jakobkmar at least can Element under XOrg on Fedora 35, unless I have misunderstood something (quite possible).

If you can't get it to run under XOrg on Fedora 35, then I think we should re-open your ticket as a distinct issue (and sorry for the runaround ;)

**Comment by jakobkmar** [comment]
> Okay - #19743 (comment) suggests that @jakobkmar at least can run Element under XOrg on Fedora 35, unless I have misunderstood something (quite possible).

You understood it correctly, for me it is only a Wayland issue, not a Fedora issue.

**Comment by damnms** [comment]
not sure what happened but the latest version from taw (fedora repository for element) does not require --disable-seccomp anymore, so i guess that has been updated and my problem is solved

**Comment by jplatte** [comment]
> Sounds promising, we will see what testing produces on other platforms.

@novocaine Has there been any progress? Just got a HiDPI notebook and Element is looking quite bad since it is drawn at low resolution and then upscaled.

**Comment by t3chguy** [comment]
https://github.com/vector-im/element-desktop/pull/299 upgrades to Electron 16 which will hopefully close this issue out.

**Comment by jplatte** [comment]
I'm not so sure. It used to work with other apps, but right now all of my Electron apps except

**Comment by jplatte** [comment]
That wasn't intentional. What I wanted to actually comment is that all my Electron apps except VSCode (for some reason) have regressed on this, and are now segfaulting at start (albeit only if `--ozone-platform=wayland` is explicitly set). "Bare" Chromium, like VSCode, works. I guess I should try building vector-im/element-desktop#299 and see whether that works.

Relevant upstream bug might be https://github.com/electron/electron/issues/31885.

**Comment by jplatte** [comment]
I tried with a dev build of element-desktop + element-web 1.9.8. It worked with my locally-installed electron from the arch linux repository while segfaulting with `node_modules/.bin/electron`. Both report version 16.0.6... Anyways I guess this means I can properly enjoy element desktop as a native desktop app once the next release is done and the element-desktop arch package updated 😊

**Comment by jplatte** [comment]
Update: Actually, the native one crashes too.. When it launches a `.asar` file instead of being given an app directory. I created a debug build of Chromium which took hours, but still didn't get the needed symbol information for finding out more about the crash 😢

**Comment by novocaine** [comment]
if you're doing your own build .. you could try hacking in the electron crash reporter?

e.g. like https://github.com/electron/electron/issues/26509#issuecomment-728267670

(on our list to integrate this properly)

**Comment by jplatte** [comment]
As long as the stack traces are all `??` entries, I don't think that would help, right? I haven't yet been able to fix that; apparently full symbols are only supported on "component builds" but I get a clang++ compile error for that, maybe I could try g++ instead.

**Comment by novocaine** [comment]
are the stack traces totally blank, or do they tell you which executable/binary the frames are in (even if the function names are missing)? if the latter, you can still see which library is responsible for the crash

**Comment by jplatte** [comment]
```
Thread 1 "electron" received signal SIGSEGV, Segmentation fault.
0x0000555557f75c1e in ?? ()
(gdb) bt
#0  0x0000555557f75c1e in  ()
#1  0x0000266e00d7c970 in  ()
#2  0xaf62bd6d4ee57b00 in  ()
#3  0x00007fffffffc840 in  ()
#4  0x00005555577cd42b in  ()
#5  0x0000266e00da6cc0 in  ()
#6  0x0000000000000018 in  ()
#7  0x0000000000000018 in  ()
#8  0x0000004554415453 in  ()
#9  0xaf62bd6d4ee57b00 in  ()
#10 0x0000266e00d7c970 in  ()
#11 0x00007fffffffc910 in  ()
#12 0x00007fffffffcbe0 in  ()
#13 0x00007fffffffcc60 in  ()
#14 0x00005555577bc7d8 in  ()
#15 0x000010c5080a0b4d in  ()
#16 0x000000000000026c in  ()
#17 0x00000000000003c0 in  ()
#18 0x0000266e00116488 in  ()
#19 0x0000026c000003c0 in  ()
#20 0x000003c00000026c in  ()
#21 0x0000266e00116240 in  ()
#22 0x0000000000000001 in  ()
#23 0x00007fffffffce30 in  ()
#24 0x0000000000000000 in  ()
```

**Comment by davidmehren** [comment]
I found a way to get stack traces without compiling electron myself:
- Download and extract the normal zip from electron's GitHub releases, e.g. https://github.com/electron/electron/releases/download/v16.0.7/electron-v16.0.7-linux-x64.zip
- Download the `-debug` variant, e.g. https://github.com/electron/electron/releases/download/v16.0.7/electron-v16.0.7-linux-x64-debug.zip
- Copy the contents of the `debug` folder inside the `-debug` zip to the location where you previously extracted the normal release zip. You should now have an `electron` binary and a significantly bigger `electron.debug` file.
- If you run `gdb ./electron` in the directory you extracted to, gdb will read debug symbols from the `electron.debug` file.
- Now run Element in gdb with `run /opt/Element-Nightly/resources/app.asar --ozone-platform=wayland`

This results in
```
(gdb) run /opt/Element-Nightly/resources/app.asar --ozone-platform=wayland
Starting program: /home/david/Downloads/electron-v16.0.7-linux-x64/electron /opt/Element-Nightly/resources/app.asar --ozone-platform=wayland
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x7fffe45cb640 (LWP 969571)]
[Detaching after fork from child process 969572]
[Detaching after fork from child process 969573]
[Detaching after fork from child process 969574]
[New Thread 0x7fffe3dca640 (LWP 969577)]
[New Thread 0x7fffe35c9640 (LWP 969578)]
[New Thread 0x7fffe2dc8640 (LWP 969579)]
[New Thread 0x7fffe25c7640 (LWP 969580)]
[New Thread 0x7fffe1dc6640 (LWP 969581)]
[New Thread 0x7fffe15c5640 (LWP 969582)]
[New Thread 0x7fffe0dc4640 (LWP 969583)]
[New Thread 0x7fffe0269640 (LWP 969584)]
[New Thread 0x7fffdfa68640 (LWP 969585)]
[New Thread 0x7fffdf267640 (LWP 969586)]
[New Thread 0x7fffdea66640 (LWP 969587)]
[New Thread 0x7fffe02c7640 (LWP 969588)]
/home/david/.config/Element-Nightly exists: yes
/home/david/.config/Riot-Nightly exists: no
[New Thread 0x7fffdd88a640 (LWP 969590)]
[New Thread 0x7fffdd055640 (LWP 969591)]
[New Thread 0x7fffdc831640 (LWP 969592)]
[New Thread 0x7fffdbfc9640 (LWP 969593)]
[New Thread 0x7fffdb7ac640 (LWP 969594)]
[New Thread 0x7fffdae12640 (LWP 969595)]
[Thread 0x7fffdae12640 (LWP 969595) exited]
[New Thread 0x7fffdae12640 (LWP 969596)]
[New Thread 0x7fffda290640 (LWP 969597)]
[Thread 0x7fffdae12640 (LWP 969596) exited]

(electron:969491): Gtk-WARNING **: 19:05:07.090: Theme file for volantes_light_cursors has no directories
[Thread 0x7fffda290640 (LWP 969597) exited]
[New Thread 0x7fffda290640 (LWP 969598)]
[New Thread 0x7fffdae12640 (LWP 969599)]
[Thread 0x7fffda290640 (LWP 969598) exited]
[New Thread 0x7fffda290640 (LWP 969600)]
[New Thread 0x7fffd99c5640 (LWP 969601)]
[Thread 0x7fffda290640 (LWP 969600) exited]
[Thread 0x7fffdae12640 (LWP 969599) exited]
[New Thread 0x7fffdae12640 (LWP 969602)]
[New Thread 0x7fffda290640 (LWP 969603)]
[Thread 0x7fffd99c5640 (LWP 969601) exited]
[Thread 0x7fffdae12640 (LWP 969602) exited]
[Thread 0x7fffda290640 (LWP 969603) exited]
[New Thread 0x7fffda290640 (LWP 969604)]
[New Thread 0x7fffda2b6640 (LWP 969605)]
[New Thread 0x7fffd99c5640 (LWP 969607)]
[New Thread 0x7fffdae12640 (LWP 969606)]
[New Thread 0x7fffd8196640 (LWP 969608)]
[New Thread 0x7fffd7975640 (LWP 969609)]
[New Thread 0x7fffd7174640 (LWP 969610)]
[New Thread 0x7fffd6973640 (LWP 969613)]
[New Thread 0x7fffd60b6640 (LWP 969614)]
[New Thread 0x7fffd5795640 (LWP 969615)]
[New Thread 0x7fffd4f94640 (LWP 969616)]
[New Thread 0x7fffd4793640 (LWP 969617)]
[New Thread 0x7fffd3f92640 (LWP 969618)]
[Detaching after fork from child process 969626]
Starting auto update with base URL: https://packages.element.io/nightly/update/
Auto update not supported on this platform
[New Thread 0x7fffd3112640 (LWP 969627)]
[969491:0125/190507.261418:ERROR:cursor_loader.cc(114)] Failed to load a platform cursor of type kNull
[969611:0125/190507.283905:ERROR:gpu_init.cc(457)] Passthrough is not supported, GL is egl, ANGLE is 
[969611:0125/190507.294176:ERROR:sandbox_linux.cc(376)] InitializeSandbox() called with multiple threads in process gpu-process.

Thread 1 "electron" received signal SIGSEGV, Segmentation fault.
std::__1::find_if<std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, base::internal::ProjectedUnaryPredicate<base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity>(base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}&, base::identity&)::{lambda(auto:1&&)#1}>(std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, base::internal::ProjectedUnaryPredicate<base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity>(base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}&, base::identity&)::{lambda(auto:1&&)#1}) (__first=..., __last=..., __pred=...) at ../../buildtools/third_party/libc++/trunk/include/algorithm:943
Download failed: Das Argument ist ungültig.  Continuing without source file ./../../buildtools/third_party/libc++/trunk/include/algorithm.
943	../../buildtools/third_party/libc++/trunk/include/algorithm: Datei oder Verzeichnis nicht gefunden.
(gdb) bt
#0  std::__1::find_if<std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, base::internal::ProjectedUnaryPredicate<base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity>(base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}&, base::identity&)::{lambda(auto:1&&)#1}>(std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, base::internal::ProjectedUnaryPredicate<base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity>(base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}&, base::identity&)::{lambda(auto:1&&)#1}) (__first=..., __last=..., __pred=...) at ../../buildtools/third_party/libc++/trunk/include/algorithm:943
#1  base::ranges::find_if<std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity, std::__1::random_access_iterator_tag>(std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, std::__1::__wrap_iter<base::internal::UncheckedObserverAdapter const*>, base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity) (first=..., last=..., pred=..., proj=...) at ../../base/ranges/algorithm.h:465
#2  base::ranges::find_if<std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter> > const&, base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity, std::__1::random_access_iterator_tag>(std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter> > const&, base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const::{lambda(auto:1 const&)#1}, base::identity) (pred=..., range=<optimized out>, proj=...)
    at ../../base/ranges/algorithm.h:483
#3  base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::HasObserver(x11::EventObserver const*) const (obs=0x112e01bb9720, this=<optimized out>) at ../../base/observer_list.h:304
#4  base::ObserverList<x11::EventObserver, false, true, base::internal::UncheckedObserverAdapter>::AddObserver(x11::EventObserver*) (this=<optimized out>, obs=0x112e01bb9720) at ../../base/observer_list.h:271
#5  x11::Connection::AddEventObserver(x11::EventObserver*) (this=0x112e000cb808, observer=0x112e01bb9720) at ../../ui/gfx/x/connection.cc:486
#6  0x000055555769c6d4 in electron::WindowStateWatcher::WindowStateWatcher(electron::NativeWindowViews*) (this=0x112e01bb9720, window=<optimized out>) at ../../electron/shell/browser/ui/x/window_state_watcher.cc:25
#7  0x000055555768dc02 in std::__1::make_unique<electron::WindowStateWatcher, electron::NativeWindowViews*>(electron::NativeWindowViews*&&) (__args=<optimized out>) at ../../buildtools/third_party/libc++/trunk/include/__memory/unique_ptr.h:725
#8  electron::NativeWindowViews::NativeWindowViews(gin_helper::Dictionary const&, electron::NativeWindow*) (this=0x112e01026900, options=..., parent=0x0) at ../../electron/shell/browser/native_window_views.cc:268
#9  0x0000555557690a3b in electron::NativeWindow::Create(gin_helper::Dictionary const&, electron::NativeWindow*) (options=..., parent=0x0) at ../../electron/shell/browser/native_window_views.cc:1652
#10 0x000055555754075e in electron::api::BaseWindow::BaseWindow(v8::Isolate*, gin_helper::Dictionary const&) (this=<optimized out>, isolate=0x112e00460000, options=...) at ../../electron/shell/browser/api/electron_api_base_window.cc:96
#11 0x000055555754f875 in electron::api::BrowserWindow::BrowserWindow(gin::Arguments*, gin_helper::Dictionary const&) (this=0x112e000cf800, args=0x7fffffff9430, options=...) at ../../electron/shell/browser/api/electron_api_browser_window.cc:33
#12 0x0000555557551b4c in electron::api::BrowserWindow::New(gin_helper::ErrorThrower, gin::Arguments*) (thrower=..., args=0x7fffffff9430) at ../../electron/shell/browser/api/electron_api_browser_window.cc:513
#13 0x000055555755217d in base::RepeatingCallback<gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)>::Run(gin_helper::ErrorThrower, gin::Arguments*) const & (this=0x112e01bb8e48, args=..., args=<optimized out>)
    at ../../base/callback.h:241
#14 gin_helper::internal::InvokeFactory<gin_helper::ErrorThrower, gin::Arguments*>(gin::Arguments*, base::RepeatingCallback<gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)> const&) (callback=..., args=<optimized out>)
    at ../../electron/shell/common/gin_helper/constructor.h:43
#15 gin_helper::internal::InvokeNew<gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)>(base::RepeatingCallback<gin_helper::WrappableBase* (gin_helper::ErrorThrower, gin::Arguments*)> const&, v8::Isolate*, gin_helper::Arguments*)
    (factory=..., isolate=<optimized out>, args=0x7fffffff9430) at ../../electron/shell/common/gin_helper/constructor.h:134
#16 0x000055555754741d in base::RepeatingCallback<void (v8::Isolate*, gin_helper::Arguments*)>::Run(v8::Isolate*, gin_helper::Arguments*) const & (this=0x7fffffff9458, args=0x112e01bb9720, args=0x0) at ../../base/callback.h:241
#17 gin_helper::Invoker<gin_helper::IndicesHolder<0ul, 1ul>, v8::Isolate*, gin_helper::Arguments*>::DispatchToCallback(base::RepeatingCallback<void (v8::Isolate*, gin_helper::Arguments*)>) (this=0x7fffffff9460, callback=...)
    at ../../electron/shell/common/gin_helper/function_template.h:230
#18 gin_helper::Dispatcher<void (v8::Isolate*, gin_helper::Arguments*)>::DispatchToCallback(v8::FunctionCallbackInfo<v8::Value> const&) (info=<optimized out>) at ../../electron/shell/common/gin_helper/function_template.h:264
#19 0x00005555585982d3 in v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) (this=0x7fffffff9548, handler=...) at ../../v8/src/api/api-arguments-inl.h:152
#20 v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) (isolate=0x112e00460000, function=..., new_target=..., fun_data=..., receiver=..., args=...) at ../../v8/src/builtins/builtins-api.cc:112
#21 0x0000555558596414 in v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) (args=..., isolate=0x112e00460000) at ../../v8/src/builtins/builtins-api.cc:138
#22 v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) (args_length=<optimized out>, args_object=<optimized out>, isolate=<optimized out>) at ../../v8/src/builtins/builtins-api.cc:130
#23 0x0000068300080eb8 in  ()
#24 0x0000000000000004 in  ()
#25 0x00007fffffff97c0 in  ()
#26 0x000000000000002a in  ()
#27 0x00007fffffff9830 in  ()
#28 0x0000068300009c2f in  ()
#29 0x00000683099a9f15 in  ()
#30 0x00000683099a9f15 in  ()
#31 0x000000000000000c in  ()
#32 0x000006830800242d in  ()
#33 0x00000683085523b1 in  ()
#34 0x0000068308551bad in  ()
#35 0x0000000000000004 in  ()
#36 0x000006830852a81d in  ()
#37 0x0000000000000026 in  ()
#38 0x00007fffffff9898 in  ()
#39 0x00000683001196fb in  ()
#40 0x0000000000000000 in  ()
(gdb) 
```

Still not complete, but at least better than nothing.

**Comment by jplatte** [comment]
Looks like this is indeed electron/electron#31885 (or electron/electron#32436 if that's not a duplicate). Either way I get the same stacktrace (on 17-beta.7 too), and the proposed fix from the first issue, electron/electron#29618, also doesn't work 🙁

I guess that PR is a good place to follow up on this.

**Comment by jplatte** [comment]
I found a workaround (instructions are arch-specific) for native-wayland Element on sway!

* Downgrade sway to 1.6.1 if already on 1.7 (add to `IgnorePkg` in `/etc/pacman.conf` to exclude from regular updates)
* Install the `electron15` package
* Add the following to `.config/electron15-flags.conf`:
  ```
  --enable-features=UseOzonePlatform --ozone-platform=wayland
  ```
* Create `/usr/local/bin/element-desktop` as
  ```bash
  #!/bin/bash
  /usr/bin/electron15 /usr/lib/element/app.asar "$@"
  ```
* `chmod +x /usr/local/bin/element-desktop`
* Profit!

**Comment by whoizit** [comment]
for me simply works this:
- Add the following to `~/.config/electron15-flags.conf`:
- `--enable-features=UseOzonePlatform --ozone-platform=wayland`
- Create `~/bin/element` as
`exec /usr/bin/electron15 /usr/lib/element/app.asar "$@"`
- `chmod +x /usr/local/bin/element-desktop`
